go/cmd/betabot/main.go

```diff
@@ -300,80 +319,94 @@ func (bot *Bot) handleUserMessageInteractionUpdated(ctx context.Context, _ *bert
 	}
 
 	var conv *Conversation
+	bot.store.RLock()
 	for i := range bot.store.Convs {
 		if bot.store.Convs[i].ConversationPublicKey == interaction.ConversationPublicKey {
-			conv = &bot.store.Convs[i]
+			conv = bot.store.Convs[i]
 		}
 	}
+	bot.store.RUnlock()
 	receivedMessage := payload.(*bertymessenger.AppMessage_UserMessage)
 	if conv != nil && conv.IsOneToOne {
-		switch {
-		case conv.Count == 0 && checkValidationMessage(receivedMessage.GetBody()):
-			conv.Count++
-			bot.saveStore()
-			time.Sleep(1 * time.Second)
+		conv.Lock()
+		defer conv.Unlock()
+		switch conv.Step {
+		case 0:
```

Comment:
I like that you made the switch easier to read, but in that case I think we should directly go to a `map[int]func()`

Reply:
I see, do you think it would be worth to also make a iota instead of using 0,1 and 2 ?

Reply:
iota will be dangerous here because we need to have step "names" however we store a "step number" in DB, so we need to stay consistent and just consider that "step 1 is after step 0" instead of "stepAddGroup is after stepAddContact"

but you can use a slice instead of a map :+1: